src/module.ts

```diff
@@ -1,25 +1,24 @@
+import * as _ from "lodash";
 import { NgModule, ModuleWithProviders, InjectionToken } from "@angular/core";
 
 import { SsvViewportMatcherDirective } from "./viewport/index";
 import { UxOptions, UX_DEFAULT_CONFIG, UX_CONFIG } from "./config";
 import { WINDOW } from "./platform/window";
+import { RecursivePartial } from "./internal/internal.model";
 
 /** @internal */
-export const _MODULE_CONFIG = new InjectionToken<UxOptions | (() => UxOptions)>(
-	"_ux-config"
-);
+export const _MODULE_CONFIG = new InjectionToken<UxOptions>("_ux-config");
 
 @NgModule({
 	declarations: [SsvViewportMatcherDirective],
 	providers: [
 		{ provide: UX_CONFIG, useValue: UX_DEFAULT_CONFIG },
-
 		{ provide: WINDOW, useFactory: _window },
 	],
 	exports: [SsvViewportMatcherDirective],
 })
 export class SsvUxModule {
-	static forRoot(config?: Partial<UxOptions> | (() => Partial<UxOptions>)): ModuleWithProviders {
+	static forRoot(config?: RecursivePartial<UxOptions> | (() => RecursivePartial<UxOptions>)): ModuleWithProviders {
```

Comment:
Added new type `RecursivePartial` to make every property within an object - optional